Added Lab Type tooltip.

jbello0909 opened the pull request:
Added a tooltip next to the lab type that describes the types of labs with the provided information.

- Used a Font Awesome icon.
- Added a React Popover component to display the desired information.

*Working feature:*

![lab type tooltip](https://user-images.githubusercontent.com/60026729/111053886-afa6a800-8435-11eb-9aa7-491497d3d3c1.gif)

jbello0909 committed:
Added lab type tooltip.

Added a tooltip next to the lab type that describes the types of labs with the provided information.

- Used a Font Awesome icon.
- Added a React Popover component to display the desired information.

jbello0909 committed:
Fixed linting errors.

- I realized there were linting errors after I pushed the initial commit. Fixed, tested and now all linting errors have been corrected.

Cliftonz commented on src/pages/LabEditor/LabEditor.tsx:
```diff
@@ -109,21 +109,21 @@ export default function LabEditor({match: {params: {moduleId, labId}}}: Props) {
-    }}>
-      <Popover.Title as="h3">Info:</Popover.Title>
+    <Popover id='popover-basic' style={{maxWidth: '750px'}}>
+      <Popover.Title as='h3'>Info:</Popover.Title>
```
I would change the title to be "Lab Types" to be more clear on what info it is displaying.

jbello0909 replied:
Done. Changed popover title from "Info" to "Lab Types".

![lab type tooltip corrected](https://user-images.githubusercontent.com/60026729/111165662-b1778500-8575-11eb-8f7f-3b6b0cbd4f9c.gif)

jasekiw commented on src/pages/LabEditor/LabEditor.tsx:
```diff
@@ -105,6 +108,24 @@ export default function LabEditor({match: {params: {moduleId, labId}}}: Props) {
   const getFieldName = (name: keyof LabForm) => name;
 
   const [selectedVm, setSelectedVm] = useState<number|undefined>();
+  const popover = (
+    <Popover id='popover-basic' style={{maxWidth: '750px'}}>
+      <Popover.Title as='h3'>Info:</Popover.Title>
+      <Popover.Content>
+        <div>- Temporary - only persist for only an hour and torn down afterwards. Their is no long term persistence with temporary labs.</div>
+        <div>
+          - Semester - these labs persist until a certain date-time is reached. 
+          Once the date-time is reach the lab is torn down is unable to be started again.
+        </div>
+        <div>- Permanent - these labs are never torn down.</div>
+      </Popover.Content>
+    </Popover>
+  );
+  const Info = () => (
+    <OverlayTrigger trigger='hover' placement='right' overlay={popover}>
+      <a><FontAwesomeIcon icon={faInfoCircle}/></a>
+    </OverlayTrigger>
+  );
```
Can you move this code into a separate component file and call it here? This will help keep this code file smaller.

jbello0909 replied:
Done. Moved popover code to a separate component and called it from `LabEditor.tsx`.

jbello0909 committed:
Completed requested changes.

- Changed popover title from "Info" to "Lab Types".
- Moved popover code to a separate component and called it from LabEditor.tsx.

Cliftonz reviewed (commented):
lgtm, @jasekiw should this component be abstracted where the component takes a title and a body so this way it is reuseable in other places?

jasekiw commented:
@Cliftonz I don't think that is necessary at the moment.
